includes/REST_API/Stories_Media_Controller.php

```diff
@@ -120,39 +120,77 @@ public function create_item( $request ) {
 		// WP_REST_Attachments_Controller doesn't allow setting an attachment as the parent post.
 		// Hence we are working around this here.
 		$parent_post = ! empty( $request['post'] ) ? (int) $request['post'] : null;
+		$original_id = ! empty( $request['original_id'] ) ? (int) $request['original_id'] : null;
 		unset( $request['post'] );
 
-		if ( ! $parent_post ) {
-			return parent::create_item( $request );
-		}
-
 		$response = parent::create_item( $request );
-		if ( is_wp_error( $response ) ) {
+		if ( is_wp_error( $response ) || ( ! $parent_post && ! $original_id ) ) {
 			return $response;
 		}
 
-		$data              = $response->get_data();
-		$post_id           = $data['id'];
-		$attachment_before = $this->get_post( $post_id );
-		if ( is_wp_error( $attachment_before ) ) {
-			return $attachment_before;
-		}
-
-		$args   = [
-			'ID'          => $post_id,
-			'post_parent' => $parent_post,
-		];
-		$result = wp_update_post( $args, true );
-		if ( is_wp_error( $result ) ) {
-			return $result;
+		$data       = $response->get_data();
+		$attachment = $this->process_post( $data['id'], $parent_post, $original_id );
+		if ( is_wp_error( $attachment ) ) {
+			return $attachment;
 		}
 
-		$data['post'] = $parent_post;
+		$new_response = $this->prepare_item_for_response( $attachment, $request );
+		$data         = $new_response->get_data();
 		$response->set_data( $data );
 
 		return $response;
 	}
 
+	/**
+	 * Process post to update attribute.
+	 *
+	 * @since 1.10.0
+	 *
+	 * @param int      $post_id Post id.
+	 * @param int|null $parent_post New post parent. Default null.
+	 * @param int|null $original_id Original id to copy data from. Default null.
+	 *
+	 * @return WP_Post|WP_Error
+	 */
+	protected function process_post( $post_id, $parent_post, $original_id ) {
```

Comment:
Another benefit of doing this server level, is devs can hook and copy their own data across, as they have the `original_id` in the request.